datafusion/optimizer/tests/optimizer_integration.rs

```diff
@@ -281,11 +278,9 @@ fn test_same_name_but_not_ambiguous() {
     let expected = "LeftSemi Join: t1.col_int32 = t2.col_int32\
     \n  Aggregate: groupBy=[[t1.col_int32]], aggr=[[]]\
     \n    SubqueryAlias: t1\
-    \n      Filter: test.col_int32 IS NOT NULL\
```

Comment:
This was actually a regression in https://github.com/apache/datafusion/pull/

Reply:
the link seems to be incomplete

Reply:
This https://github.com/apache/datafusion/pull/12348 , I think I move the fix out of this PR, so it can be reviewed separately

Reply:
https://github.com/apache/datafusion/pull/12404